Aardvark/Aardvark.swift

```diff
@@ -0,0 +1,157 @@
+            
+            UIApplication.observingKeyWindowNotifications = true
+        }
+        
+        return bugReportingGestureRecognizer
+    }
+    
+    @nonobjc func removeBugReporter(bugReporter: ARKBugReporter) {
+        if let gestureRecognizerToRemove: UIGestureRecognizer = UIApplication.bugReporterToGestureRecognizerMap.objectForKey(bugReporter) as? UIGestureRecognizer {
+            gestureRecognizerToRemove.view?.removeGestureRecognizer(gestureRecognizerToRemove)
+            
+            UIApplication.bugReporterToGestureRecognizerMap.removeObjectForKey(bugReporter)
+        }
+    }
+    
+    @objc(ARK_didFireBugReportGestureRecognizer:) func didFireBugReportGestureRecognizer(bugReportRecognizer: UIGestureRecognizer) {
+        guard bugReportRecognizer.state == .Began else
+        { return }
+        
+        var bugReporters = [ARKBugReporter]()
+        for bugReporter in UIApplication.bugReporterToGestureRecognizerMap.keyEnumerator() {
+            guard let bugReporter = bugReporter as? ARKBugReporter where !bugReporters.contains({ $0 === bugReporter }) else
+            { continue }
+            guard let recognizerForBugReport = UIApplication.bugReporterToGestureRecognizerMap.objectForKey(bugReporter) as? UIGestureRecognizer where recognizerForBugReport == bugReportRecognizer else
+            { continue }
+            
+            bugReporters.append(bugReporter)
+        }
+        
+        guard bugReporters.count > 0 else
+        { return }
+        
+        for bugReporter in bugReporters {
+            bugReporter.composeBugReport()
+        }
+    }
+    
+    @objc(ARK_windowDidBecomeKeyNotification:) func windowDidBecomeKeyNotification(notification: NSNotification) {
+        guard let window = notification.object as? UIWindow else
+        { return }
+        guard let gestureRecognizersEnumerator = UIApplication.bugReporterToGestureRecognizerMap.objectEnumerator() else
+        { return }
+        
+        for gestureRecognizer in gestureRecognizersEnumerator {
+            if let gestureRecognizer = gestureRecognizer as? UIGestureRecognizer {
+                window.addGestureRecognizer(gestureRecognizer)
+            }
+        }
+    }
+    
+    @objc(ARK_windowDidResignKeyNotification:) func windowDidResignKeyNotification(notification: NSNotification) {
+        guard let window = notification.object as? UIWindow else
+        { return }
+        guard let gestureRecognizersEnumerator = UIApplication.bugReporterToGestureRecognizerMap.objectEnumerator() else
+        { return }
+        
+        for gestureRecognizer in gestureRecognizersEnumerator {
+            if let gestureRecognizer = gestureRecognizer as? UIGestureRecognizer {
+                window.removeGestureRecognizer(gestureRecognizer)
+            }
+        }
+    }
+}
+
+// MARK: Private Functions
+
+private func noteImproperAPIUse(message: String) {
```

Comment:
Thoughts on the name here? Was tempted to call this `ARKNoteImproperAPIUse`... but since it's a private method it didn't seem necessary.